portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -26,55 +24,83 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.time.LocalDateTime;
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.google.sps.utils.Constants.COMMENT_COMMENTER;
+import static com.google.sps.utils.Constants.COMMENT_CONTENT;
+import static com.google.sps.utils.Constants.COMMENT_KEY;
+import static com.google.sps.utils.Constants.COMMENT_TIMESTAMP;
+
 /** Servlet that handles getting and posting comment content. */
 @WebServlet("/comment")
 public final class CommentServlet extends HttpServlet {
-  private final List<Comment> comments = new ArrayList<>();
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(this.comments);
+    // Loads the comment from Datastore
+    List<Comment> comments = getComments();
 
-    // Send the JSON as the response
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
+    // Converts into json form
+    String json = convertToJsonUsingGson(comments);
+
+    // Sends the JSON as the response
+    sendJsonResponse(response, json);
   }
 
+  /** Loads the comment from Datastore */
+  private List<Comment> getComments() {
+    Query query = new Query(COMMENT_KEY).addSort(COMMENT_TIMESTAMP, Query.SortDirection.DESCENDING);
```

Comment:
Can we abstract these datastore load logic out to a separate method? Or maybe you can create a CommentDataStore class and have it provide load/store comments? This way we isolate the processing logic from the persistent logic.

Reply:
Fixed